class-components/src/App.tsx

```diff
@@ -0,0 +1,87 @@
+import { Component } from 'react';
+
+import './App.css';
+import Search from './components/Search/Search';
+
+import { IAppState, ISingleResult } from './types/SearchTypes';
+import { BASE_URL, LS_QUERY } from './constants';
+import Results from './components/Results/Results';
+
+class App extends Component<object, IAppState> {
+  constructor(props: object) {
+    super(props);
+    this.state = {
+      results: [],
+      hasError: false,
+      isLoading: false,
+    };
+  }
+
+  componentDidMount() {
+    const query = localStorage.getItem(LS_QUERY);
+    if (query) {
+      this.fetchData(query);
+    } else {
+      this.fetchData('');
+    }
+  }
+
+  fetchData = async (query: string) => {
+    this.setState({ isLoading: true });
+    try {
+      const resp = await fetch(`${BASE_URL}/people/?search=${query}`);
+      if (!resp.ok) {
+        throw new Error('Fetch response failed');
+      }
+      const data = await resp.json();
+      this.setState({
+        results: data.results.map((item: ISingleResult) => ({
+          name: item.name,
+          birth_year: item.birth_year,
+          gender: item.gender,
+        })),
```

Comment:
you absolutely should not do anything like this in setState 
Instead do: 
```suggestion
const results = data.results.map(() => ());
      this.setState({
        results,
```